portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,38 +30,35 @@
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.FetchOptions;
+
 
 /** Servlet that returns data for my portfolio. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List comments = new ArrayList<Comment>();
   private static final int DEFAULT_NUM_COMMENTS = 10;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Load from datastore
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    PreparedQuery results = datastore.prepare(query);
+    List comments = new ArrayList<Comment>();
 
     int commentsMaxNum = DEFAULT_NUM_COMMENTS;
     try {
       commentsMaxNum = new Integer(request.getParameter("comments-num"));
     } catch (NumberFormatException e) {
       System.err.println("Num of comments wasn't well defined.");
     }
-    
-    comments.clear();
-    for (Entity entity : results.asIterable()) {
-      if (commentsMaxNum <= 0) {
-          break;
-      }
+
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(commentsMaxNum));
```

Comment:
This is a great change - oneline-ing the expression. 
Please pay attention not to pass the 80-chars limit (I don't know if one was defined for you, but this is common practice).

Reply:
I went ahead and committed this change. Please take a look and then merge the PR.

Reply:
👍